SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -77,11 +77,33 @@ def add? {w : Nat} (x y : BitVec w) : IntW w :=
 @[simp_llvm_option]
 theorem add?_eq : LLVM.add? a b  = .some (BitVec.add a b) := rfl
 
+structure AdditionFlags where
+  nsw : Bool := false
+  nuw : Bool := false
+  deriving DecidableEq
+
+/-- Does the signed addition x + y overflow? -/
+def AddSignedWraps? (x y : BitVec w) : Bool := (x.toInt + y.toInt) < -(2^(w-1)) ∨ (x.toInt + y.toInt) ≥ 2^w
```

Comment:
Please have a look at https://leanprover-community.github.io/contribute/naming.html;
as a `Bool`-valued `def` this should be `lowerCamelCase`d. 

However, this really should be a `Prop` instead. In fact, if you `#print` the def you'll notice the rhs is actually a Prop and it's being coerced into a `Bool` by `decide`